include/gamepad/event_handler.hpp

```diff
@@ -24,33 +24,33 @@ template <typename Key, typename... Args> class EventHandler {
          *
          * @param key The listener key (this must be a unique key value)
          * @param func The function to run when this event is fired
-         * @return true The listener was successfully added
-         * @return false The listener was NOT successfully added (there is already a listener with the same key)
+         * @return 0 The listener was successfully added
+         * @return UINT32_MAX The listener was NOT successfully added (there is already a listener with the same key)
          */
-        bool addListener(Key key, Listener func) {
+        uint32_t addListener(Key key, Listener func) {
             std::lock_guard lock(m_mutex);
-            if (std::find(m_keys.begin(), m_keys.end(), key) != m_keys.end()) return false;
+            if (std::find(m_keys.begin(), m_keys.end(), key) != m_keys.end()) return UINT32_MAX;
```

Comment:
Functions should return `INT32_MAX` when errno is set, so it does not differ from PROS and other LemLib projects

```suggestion
            if (std::find(m_keys.begin(), m_keys.end(), key) != m_keys.end()) return INT32_MAX;
```